src/azul/plugins/repository/tdr/__init__.py

```diff
@@ -176,6 +182,21 @@ def create(cls, catalog: CatalogName) -> 'RepositoryPlugin':
     def sources(self) -> AbstractSet[TDRSourceName]:
         return self._sources
 
+    def list_sources(self,
+                     authentication: Optional[Authentication]
+                     ) -> JSONs:
+        if isinstance(authentication, OAuth2):
+            sources = TDRClient.with_user_credentials(authentication).list_snapshots()
+            catalog_sources = {source.name for source in self.sources}
+            return [source for source in sources if source['sourceName'] in catalog_sources]
```

Comment:
Use set.intersection?

Reply:
I don't think `set.intersection` can be efficiently applied here because `sources` is a `Sequence[dict]` and `catalog_sources` is a `Sequence[str]`. I've renamed `catalog_sources` to `catalog_source_names` to make this more clear.